src/ports/postgres/modules/deep_learning/madlib_keras_predict.py_in

```diff
@@ -112,6 +114,9 @@ def predict(schema_madlib, model_table, test_table, id_col,
         class_values, intermediate_col, pred_col_name,
         pred_col_type, is_response, MODULE_NAME)
 
+    segment_id = -1 if is_platform_pg() else '{0}.gp_segment_id'.format(
```

Comment:
why do we need to qualify the segment_id with the table name?

Reply:
The query that this gets used in selects from both `{test_table}` and `{model_table}`. `gp_segment_id` is an implicit column in both these tables that would lead to ambiguous column name.

Reply:
Makes sense. Maybe we should create a function named `get_segment_id` which will replace all instances of this if check ? It will make our code less error prone to the situation you described.

Reply:
Sure, but the only other instance is in fit, and there we don't have to prefix it with a table name.
Btw, the value of this variable for postgres makes no difference (it could be -1 or any other integer), since we do not infer it is postgres using this value anywhere else in the code. It just shouldn't be `gp_segment_id` for postgres.

Reply:
Realized that if we use the refactored function, we will have to call it
twice in fit function, once for train table and once for validation
table, since the function prefixes the table name for GPDB. I think it's
fine the way it was now, we can refactor it later if the use for this check
gets more ubiquitous. So reverted the refactoring commit, and will keep
it this way for now.